### PR TITLE
[ENH] Add fn kwargs and tests for sklearn metrics

### PR DESCRIPTION
**Pull request by paxcema**
This PR:
- cleans up some signatures to avoid unsupported arguments making their way through
- changes main entry point so that keyword arguments can be passed to any accuracy metric in a general way. This is useful to control behavior of external functions like those found in `scikit-learn.metrics`.
- adds tests to check that `sklearn.metrics` can be customized through keyword arguments correctly

### CODE REVIEW

**Comment by paxcema** [comment]
@pedrofluxa ^ FYI
